### PR TITLE
Auto mtu

### PR DESCRIPTION
**Pull request by psmithcrl**

### CODE REVIEW

File: source/LibMultiSense/details/public.cc [hunk 1]
```diff
@@ -1343,6 +1343,56 @@ Status impl::getMtu(int32_t& mtu)
...
+    
+    if (m_sensorVersion.firmwareVersion <= 0x0202)
+        return Status_Unsupported;
+
+    while (bisections < 7){
+        std::cout << "Testing MTU " << cur_mtu << std::endl;
+        wire::SysTestMtuResponse resp;
+        status = waitData(wire::SysTestMtu(cur_mtu), resp, 0.1, 1);
+        if ((Status_Ok == status) && (cur_mtu == MAX_MTU_SIZE)){
+            std::cout << "Max MTU negeotiated on first try!\n";
```
Comment (mattalvarado) [inline source/LibMultiSense/details/public.cc:1365]: Do we need this print?

File: source/LibMultiSense/details/public.cc [hunk 2]
```diff
@@ -1343,6 +1343,56 @@ Status impl::getMtu(int32_t& mtu)
...
+    Status status = Status_Ok;
+
+    //
+    // v2.2 and older do not support testing MTU
+    
+    if (m_sensorVersion.firmwareVersion <= 0x0202)
+        return Status_Unsupported;
+
+    while (bisections < 7){
+        std::cout << "Testing MTU " << cur_mtu << std::endl;
```
Comment (mattalvarado) [inline source/LibMultiSense/details/public.cc:1361]: Do we need this print?

**Review by mattalvarado — APPROVED** [review]
This is awesome! Thanks for adding this!

File: source/LibMultiSense/details/public.cc (hunk 2 above)
Comment (dougbalish1) [reply source/LibMultiSense/details/public.cc:1361]: No, just for visibility in testing. Will remove before merge

**Comment by mattalvarado** [comment]
> @mattalvarado Want me to do something like
> 
> ```
>     uint32_t    mtu            = 0;
>     ...
>     if (mtu >= 1500)
>         status = channelP->setMtu(mtu);
>     else 
>         status = channelP->setBestMtu();
> ```
> 
> In the utils too, or leave it at 1500?

Yeah, I like this approach!

**Pull request merged**